Tests/tests/test_RDKShell.cpp

```diff
@@ -0,0 +1,1041 @@
+                                                                                "\"enable\": true"
+                                                                                "}"), response));
+   ON_CALL(compositormock, enableInputEvents(::testing::_, ::testing::_))
+            .WillByDefault(::testing::Invoke(
+                [](const std::string& client, bool enable){
+		      EXPECT_EQ(client, string("searchanddiscovery"));
+		      EXPECT_EQ(enable, true);
+                      return true;
+                   }));
+
+   EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("enableInputEvents"), _T("{"
+                                                                                "\"clients\": ["
+                                                                                "    \"searchanddiscovery\""
+                                                                                "],"
+                                                                                "\"enable\": true"
+                                                                                "}"), response));
+}
+
+TEST_F(RDKShellTest, getClients)
+{
+      ON_CALL(compositormock, getClients(::testing::_))
+            .WillByDefault(::testing::Invoke(
+                [](std::vector<std::string>& clients){
+                      clients.push_back("org.rdk.Netflix");
+                      clients.push_back("org.rdk.RDKBrowser2");
+                      clients.push_back("Test1");
+		      clients.push_back("Test2");
+                      return true;
+                   }));
+
+     EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getClients"), _T("{}"), response));
+     EXPECT_EQ(response, string("{"
+        "\"clients\":["
+            "\"org.rdk.Netflix\","
+	    "\"org.rdk.RDKBrowser2\","
+	    "\"Test1\","
+	    "\"Test2\""
+        "],"
+        "\"success\":true"
+    "}"));	
+}
+
+TEST_F(RDKShellTest, keyRepeatConfig)
+{
+   EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("keyRepeatConfig"), _T("{}"), response));
+   ON_CALL(compositormock, setKeyRepeatConfig(::testing::_, ::testing::_, ::testing::_))
+            .WillByDefault(::testing::Invoke(
+                [](bool enabled, int32_t initialDelay, int32_t repeatInterval){
+                      EXPECT_EQ(enabled, true);
+		      EXPECT_EQ(initialDelay, 500);
+		      EXPECT_EQ(repeatInterval, 250);
+                      return true;
+                }));
+
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("keyRepeatConfig"), _T("{"
+                                                                                      "\"enabled\":true,"
+										      "\"initialDelay\":500,"
+										      "\"repeatInterval\":250}"), response));
+}
+
+TEST_F(RDKShellTest, resetinactivity)
+{
+	ON_CALL(compositormock, resetInactivityTime())
+            .WillByDefault(::testing::Invoke(
+                [&](){
+	          RdkShell::seconds();
```

Comment:
what's RdkShell::seconds(); for?
if resetInactivityTime() is important please use EXPECT_CALL